### PR TITLE
[FIX] agreement_legal: Issue #331

### LINKED ISSUE #331: [12.0][BUG] agreement_legal not installable
Hello,

I have just spotted that module `agreement_legal` is not installable: [Travis Log](https://travis-ci.org/OCA/contract/jobs/532814199#L1573)

### PR DESCRIPTION
**Pull request by max3903**
Fix #331

### CODE REVIEW

File: agreement_legal/models/agreement.py [hunk 1]
```diff
@@ -389,6 +388,8 @@ def create(self, vals):
             vals["code"] = self.env["ir.sequence"].next_by_code(
                 "agreement"
             ) or _("New")
+        vals["stage_id"] = \
```
Comment (pedrobaeza) [inline agreement_legal/models/agreement.py]: This should be conditional: `if not vals.get('stage_id'):`

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by pedrobaeza** [comment]
`_default_stage_id` is useless now. And you should also protect `self.env.ref` calls because you can remove the stage in UI and you'll get a code error in that case.

**Review by patrickrwilson — APPROVED** [review]

**Comment by max3903** [comment]
@katyukha Can you review this fix?

**Review by ygol — APPROVED** [review]

**Review by osi-scampbell — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@max3903 why have you ignored my comments at https://github.com/OCA/contract/pull/333#issuecomment-493615492 ?

**Comment by max3903** [comment]
@pedrobaeza I removed `_default_stage_id` like you suggested. Concerning the protection of `self.env.ref`, I guess I didn't understand what you meant. What did you expect?

**Pull request merged**